subcommand/get-consul-client-ca/command_test.go

```diff
@@ -0,0 +1,388 @@
+	"time"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/freeport"
+	"github.com/hashicorp/consul/sdk/testutil"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+	"github.com/hashicorp/consul/tlsutil"
+	"github.com/mitchellh/cli"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRun_FlagsValidation(t *testing.T) {
+	t.Parallel()
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI: ui,
+	}
+
+	exitCode := cmd.Run([]string{
+		"-output-file", "",
+	})
+	require.Equal(t, 1, exitCode)
+	require.Contains(t, ui.ErrorWriter.String(), "-output-file must be set")
+}
+
+// Test that in the happy case scenario
+// we retrieve the CA from Consul and
+// write it to a file
+func TestRun(t *testing.T) {
+	t.Parallel()
+	outputFile, err := ioutil.TempFile("", "ca")
+	require.NoError(t, err)
+
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI: ui,
+	}
+
+	// start the test server
+	a, err := testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
+		c.Connect = map[string]interface{}{
+			"enabled": true,
+		}
+	})
+	require.NoError(t, err)
+	defer a.Stop()
+
+	// run the command
+	exitCode := cmd.Run([]string{
+		"-server-addr", a.HTTPAddr,
+		"-output-file", outputFile.Name(),
+	})
+	require.Equal(t, 0, exitCode)
+
+	client, err := api.NewClient(&api.Config{
+		Address: a.HTTPAddr,
+	})
+	require.NoError(t, err)
+
+	// get the actual root ca cert from consul
+	roots, _, err := client.Agent().ConnectCARoots(nil)
+	require.NoError(t, err)
+	require.NotNil(t, roots)
+	require.NotNil(t, roots.Roots)
+	require.Len(t, roots.Roots, 1)
+	require.True(t, roots.Roots[0].Active)
+	expectedCARoot := roots.Roots[0].RootCertPEM
+
+	// read the file contents
+	actualCARoot, err := ioutil.ReadFile(outputFile.Name())
+	require.NoError(t, err)
+	require.Equal(t, expectedCARoot, string(actualCARoot))
+}
+
+// Test that if the Consul server is not available at first,
+// we continue to poll it until it comes up.
+func TestRun_ConsulServerAvailableLater(t *testing.T) {
+	t.Parallel()
+	outputFile, err := ioutil.TempFile("", "ca")
+	require.NoError(t, err)
+
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI: ui,
+	}
+
+	randomPorts := freeport.MustTake(6)
+
+	// Start the command asynchronously
+	exitCode := -1
+	go func() {
+		exitCode = cmd.Run([]string{
+			"-server-addr", fmt.Sprintf("http://127.0.0.1:%d", randomPorts[1]),
+			"-output-file", outputFile.Name(),
+		})
+		require.Equal(t, 0, exitCode)
+	}()
+
+	// start the test server
+	time.Sleep(500 * time.Millisecond)
+	a, err := testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
+		c.Ports = &testutil.TestPortConfig{
+			DNS:     randomPorts[0],
+			HTTP:    randomPorts[1],
+			HTTPS:   randomPorts[2],
+			SerfLan: randomPorts[3],
+			SerfWan: randomPorts[4],
+			Server:  randomPorts[5],
+		}
+		c.Connect = map[string]interface{}{
+			"enabled": true,
+		}
+	})
+	require.NoError(t, err)
+	defer a.Stop()
+
+	// wait for command to exit
+	retry.Run(t, func(r *retry.R) {
```

Comment:
It won't. By default it will timeout after 7 seconds
```
func Run(t Failer, f func(r *R)) {
	run(DefaultFailer(), t, f)
}
func DefaultFailer() *Timer {
	return &Timer{Timeout: 7 * time.Second, Wait: 25 * time.Millisecond}
}
```